sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/TypeCoercion.scala

```diff
@@ -120,11 +120,11 @@ object TypeCoercion {
    */
   private def findCommonTypeForBinaryComparison(
       dt1: DataType, dt2: DataType, conf: SQLConf): Option[DataType] = (dt1, dt2) match {
-    // We should cast all relative timestamp/date/string comparison into string comparisons
+    case (StringType, DateType) => Some(DateType)
```

Comment:
I have removed these 2 cases but the added test still completes successfully. We need a few tests that fail without the changes in `findCommonTypeForBinaryComparison`.

Reply:
It is explained by https://github.com/apache/spark/pull/24567/files#issuecomment-491579283

Reply:
Done. 
Thanks for pointing it out. It simplifies the code a lot.